app/javascript/components/forms/LocationFields.vue

```diff
@@ -5,26 +5,53 @@
       label="Street address"
       custom-class="is-medium"
     >
-      <b-input :name="withPrefix('street_address')" :value="street_address" />
+      <b-input :name="withPrefix('street_address')" v-model="streetAddress" />
     </b-field>
 
-    <!-- TODO: replace this with a dropdown -->
-    <b-input :name="withPrefix('location_type')" type="hidden" value="1" />
+    <b-field
+      v-if="streetAddress.length"
+      :label-for="withPrefix('location_type')"
+      label="Address type"
+      custom-class="required-field is-medium"
+    >
+      <b-select
+        :name="withPrefix('location_type')"
+        :value="location_type.id"
+        placeholder="Type"
+        required
```

Comment:
Is this only required if a location is present?

Reply:
address is empty, but still getting location_type required ?

Reply:
Oops! Good catch. Pushed up a [commit](26b84ef) to accommodate empty location. Also threw in that [`find_or_new`](39067e4) method :).